### PR TITLE
[Feat] add category filter on review page

### PR DESCRIPTION
**Pull request by iknowright**
## Types of changes
**Thanks for sending a pull request! Please fill in the following content to let us know better about this change.**
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
**Describe what the change is**
Add category filed to proposal table. When you click on a category, the list of proposals will be filtered by the category you selected.

## Steps to Test This Pull Request
1. Go to review page
2. Click on category filed

## Expected behavior
Show the proposal list which was be filtered.

## Related Issue
If applicable, refernce to the issue related to this pull request.

## More Information
**Screenshots**

**Additional context**
Add any other context about the problem here. You may also want to refer
to [how to wirte the perfect pull request](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)

https://user-images.githubusercontent.com/31035650/162525223-95f100f8-a7d8-4bef-a44b-37056f6d0a64.mov

### CODE REVIEW

File: src/reviews/views.py [hunk 1]
```diff
@@ -132,9 +148,12 @@ def get_context_data(self, **kwargs):
             ),
             'vote': vote_mapping,
             'ordering': self.ordering,
+            'category': self.category,
             'query_string': self.request.GET.urlencode(),
             **self.reviews_state._asdict(),
+            **self.get_category_metrics(context),
         })
+        print(context)
```
Comment (josix) [inline src/reviews/views.py:156]: I guess this is used for debugging? could you help remove it, thanks!

File: src/templates/default/reviews/_includes/proposal_table.html [hunk 2]
```diff
@@ -33,6 +33,21 @@
...
+			<select onchange="javascript:location.href = this.value;">
+				{% if not category %}
+				<option value="?order=count">ALL</option>
+				{% else %}
+				<option value="?order=count&category={{ category }}">{{ category }}</option>
+				<option value="?order=count">ALL</option>
+				{% endif %}
+				{% for category_option in category_options %}
+				{% if category != category_option %}
+				<option value="?order=count&category={{ category_option }}">{{ category_option }}</option>
```
Comment (josix) [inline src/templates/default/reviews/_includes/proposal_table.html:46]: **issue:** the value `order=count` here will cause the proposals ordered only by the count after selecting the category even the user has chosen to ordered by other field like `lang`, `count`, etc. I guess we need to pass the query parameter from URL.

**Review by josix — CHANGES_REQUESTED** [review]

File: src/reviews/views.py (hunk 1 above)
Comment (iknowright) [reply src/reviews/views.py]: Resolved in daa9f1bfcba8ff46b949f461027138e7a37e664d

File: src/templates/default/reviews/_includes/proposal_table.html (hunk 2 above)
Comment (iknowright) [reply src/templates/default/reviews/_includes/proposal_table.html]: Resolved in daa9f1bfcba8ff46b949f461027138e7a37e664d
use a trick to retain both order={}&category={} query params.

**Review by josix — APPROVED** [review]
LGTM, Sorry that I didn't notice that there is an update for the last review.

**Pull request merged**